lib/literal.ts

```diff
@@ -96,10 +96,22 @@ class LiteralTranspiler extends base.TranspilerBase {
         this.emit('new RegExp (');
         this.emit('r\'');
         var regExp = (<ts.LiteralExpression>node).text;
-        regExp = regExp.substring(1, regExp.length - 1);  // cut off /.../ chars.
+        var slashIdx = regExp.lastIndexOf('/');
+        var flags = regExp.substring(slashIdx + 1);
+        regExp = regExp.substring(1, slashIdx);  // cut off /.../ chars.
         regExp = regExp.replace('\'', '\' + "\'" + r\'');  // handle nested quotes by concatenation.
         this.emitNoSpace(regExp);
         this.emitNoSpace('\'');
+        if (flags.indexOf('g') === -1) {
+          // Dart RegExps are always multiline, so JS regexps must use 'g' so that semantics match.
```

Comment:
g is still not "multiline" but "global" :)

I'm unsure about this if branch, depends on what we do for `exec` ? I guess it good for now.